Task-WebRTC.md

```diff
@@ -0,0 +1,370 @@
+of the [WebRTC specification](https://www.w3.org/TR/webrtc/).
+
+## Wrapped Data Channel
+
+This protocol adds another security layer for WebRTC data channels as 
+we want our protocol to sustain broken (D)TLS environments. A data 
+channel that uses this security layer will be called *Wrapped Data 
+Channel*. Note that the user MAY or MAY NOT choose to use wrapped data 
+channels for its purpose. However, the signalling channel MUST be 
+handed over to a wrapped data channel.
+
+# Task Protocol Name
+
+The following protocol name SHALL be used for task negotiation:
+
+`v0.webrtc.tasks.saltyrtc.org`
+
+TODO: Switch to `v1` as soon as the spec has been reviewed.
+
+# Detecting the Maximum Message Size
+
+For now, all implementations SHALL use the value `16384` which seems 
+to be the highest amount of kilobytes that can be applied for portable 
+WebRTC data channel communication. An implementation MAY use another 
+value if it can guarantee delivery and reception for messages of that 
+size. A value of `0` indicates that the implementation is able to 
+handle messages of arbitrary length (hooray!).
+
+# Task Data
+
+This task makes use of the *data* field of the 'auth' messages 
+described in the 
+[SaltyRTC protocol specification](./Protocol.md#auth-message).  
+The *Outgoing* section describes what the data of this task SHALL 
+contain and how it MUST be generated. Whereas the *Incoming* section 
+describes how the task's data from the other client's 'auth' message 
+MUST be validated and potentially stored.
+
+## Outgoing
+
+The task's data SHALL be a dictionary/an object containing the 
+following items:
+
+* A client MUST set the *cookie_2* field to 16 cryptographically 
+  secure random bytes. It SHALL be different to the cookie the client 
+  currently uses, the other client currently uses and, for initiators, 
+  the *cookie_2* the responder has sent in its task's data.
+* The *exclude* field MUST contain a list/an array of WebRTC data 
+  channel ids (non-negative integers) that SHALL not be used for the 
+  signalling channel. This list MUST be available to be set from user 
+  applications that use specific data channel ids.
+* The *max_size* field MUST be set to the value described by the 
+  *Detecting the Maximum Message Size* section.
+
+## Incoming
+
+A client who receives the task's data from the other peer MUST do the 
+following checks:
+
+* A client SHALL validate that the *cookie_2* field's value contains
+  16 bytes and is different to the other client's cookie it currently 
+  uses, different to the client's current cookie and different to the 
+  client's upcoming cookie.
+* The *exclude* field MUST contain a list/an array of WebRTC data 
+  channel IDs (non-negative integers) that SHALL not be used for the 
+  signalling channel. The client MUST update its internal list of 
+  excluded data channel ids by new values of the other client's list. 
+  The resulting list MUST be stored sorted in ascending order.
```

Comment:
Agree. It's necessary to determine the lowest possible data channel id not excluded by both clients.